csharp/src/Drivers/Apache/Hive2/HiveServer2Parameters.cs

```diff
@@ -0,0 +1,67 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+
+namespace Apache.Arrow.Adbc.Drivers.Apache.Hive2
+{
+    public static class HiveServer2DataTypeConversionConstants
```

Comment:
Are we really going to inflict this symbol on the public API? :(

Reply:
How much of what's in this file actually needs to be public? Isn't it sufficient to expose the None and Scalar constants? Perhaps these could be moved to a more neutral name?

SupportedList is a concept that the ADBC API desperately needs, but doesn't generically support. Can we wait for a more generic "supported options" API before doing something like this?

Reply:
Replaced with `DataTypeConversionOptions`. The namespace with take care of the rest.